File: _legacy_expo/src/config/legal.ts
// Single source of truth for legal URLs — referenced from the paywall
// (App Store guideline 3.1.2), onboarding, and the message-report action
// (guideline 1.2 UGC/AI-content expectations: reporting + published contact).
export const PRIVACY_URL = 'https://github.com/VedSoni-dev/privateaiapp/blob/main/PRIVACY.md';
export const TERMS_URL = 'https://github.com/VedSoni-dev/privateaiapp/blob/main/TERMS.md';

const REPO = 'VedSoni-dev/privateaiapp';

// Reuses the public issue tracker already named as the contact channel in
// PRIVACY.md, rather than inventing a separate inbox — prefilled so the
// reporter doesn't have to explain what "report" means.
export function reportContentUrl(messageText: string): string {
  const title = 'Reported AI response';
  const snippet = messageText.length > 500 ? `${messageText.slice(0, 500)}…` : messageText;
  const body = `Flagged from the app — please review:\n\n---\n${snippet}\n---\n\n(What was wrong with it?)`;
  return `https://github.com/${REPO}/issues/new?title=${encodeURIComponent(title)}&body=${encodeURIComponent(body)}`;
}

File: PrivateAI/Models/Legal.swift
import Foundation

enum Legal {
    static let privacyURL = URL(string: "https://github.com/VedSoni-dev/privateaiapp/blob/main/PRIVACY.md")!
    static let termsURL = URL(string: "https://github.com/VedSoni-dev/privateaiapp/blob/main/TERMS.md")!
    static let appStoreURL = URL(string: "https://apps.apple.com/app/id6785089361")!
    static let appStoreDisplay = "apps.apple.com/app/id6785089361"

    static func reportContentURL(messageText: String) -> URL {
        let title = "Reported AI response"
        let snippet = messageText.count > 500 ? String(messageText.prefix(500)) + "…" : messageText
        let body = "Flagged from the app — please review:\n\n---\n\(snippet)\n---\n\n(What was wrong with it?)"
        var comps = URLComponents(string: "https://github.com/VedSoni-dev/privateaiapp/issues/new")!
        comps.queryItems = [
            URLQueryItem(name: "title", value: title),
            URLQueryItem(name: "body", value: body),
        ]
        return comps.url!
    }
}
